src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/common/ParseRequestService.java

```diff
@@ -91,13 +91,17 @@ private void setCorrelationId(Request request) {
         request.setCorrelationId(correlationInterceptorUtil.preHandle(httpServletRequest));
     }
 
-    public String getCorrelationId() {
+    public String getRequestCorrelationId() {
         HttpServletRequest httpServletRequest = ((ServletRequestAttributes) RequestContextHolder
             .currentRequestAttributes())
             .getRequest();
         return correlationInterceptorUtil.preHandle(httpServletRequest);
     }
 
+    public String getContextCorrelationId() {
+        return correlationInterceptorUtil.getContextCorrelationId();
+    }
+
```

Comment:
gets the correlation Id from MDC rather than the request, removing the possibility of two different id's being generated.